analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count.Rmd

```diff
@@ -0,0 +1,375 @@
+                                      grepl("C4113612-Tumor", sample_id) ~ "Diagnosis",
+                                      grepl("Primary Tumor|Initial CNS Tumor", tumor_descriptor) ~ "Diagnosis",
+                                      grepl("Progressive Disease Post-Mortem", tumor_descriptor) ~ "Deceased",
+                                      TRUE ~ tumor_descriptor),
+         match_id = paste(tumor_descriptor, sample_id, sep = "_"),
+         DNA = case_when(grepl("WGS|WXS", experimental_strategy) ~ "DNA",
+                         grepl("RNA-Seq", experimental_strategy) ~ "no"),
+         match_id_DNA = paste(Kids_First_Participant_ID, tumor_descriptor, DNA, sep = "_")) %>%
+  distinct(match_id_DNA, .keep_all = TRUE) # filter out
+
+print(pbta %>% count(tumor_descriptor))
+
+kids_pbta_samples <- length(unique(pbta$Kids_First_Participant_ID))
+```
+## How many samples have timepoint information?
+There are `r kids_pbta_samples` patient cases with timepoint information. 
+
+## Filter based on both genomic and transcriptomic assays
+Let's select all patient samples containing both genomic and transcriptomic assays.
+```{r assays, echo=TRUE}
+assays <- pbta %>%
+  dplyr::select(Kids_First_Participant_ID, tumor_descriptor, match_id, experimental_strategy, cancer_group, DNA, match_id_DNA) %>%
+  unique() %>%
+  arrange(Kids_First_Participant_ID, experimental_strategy)
+
+# we perform join here using the Kids_First_Participant_ID and match_id
+paired_assays_all <- assays %>%
+  group_by(Kids_First_Participant_ID, match_id, tumor_descriptor, cancer_group) %>%
+  summarise(experimental_strategy = str_c(experimental_strategy, collapse = ";")) %>% 
+  mutate(experimental_strategy_sum = case_when(grepl("RNA-Seq", experimental_strategy) & grepl("WGS|WXS", experimental_strategy) ~ "Both",
+                                               grepl("WXS|WGS", experimental_strategy) ~ "DNA",
+                                               grepl("RNA-Seq", experimental_strategy) ~ "RNA"))
+```
+
+# Number of samples for each assay
+First, we’ll examine how many of each type of assay we have (tumors only). This information is stored in the experimental_strategy column.
+```{r}
+paired_assays_all %>%
+  group_by(experimental_strategy) %>%
+  tally() %>%
+  arrange(desc(n)) %>% 
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+
+paired_assays_all %>%
+  group_by(experimental_strategy_sum) %>%
+  tally() %>%
+  arrange(desc(n)) %>% 
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+```
+
+# Diagnosis vs. any other timepoint (Recurrence, Progressive etc)
+We’ll look at the breakdown of the tumor_descriptor column, separating the genomic from transcriptomic assays.
+```{r}
+paired_assays_all %>%
+  arrange(Kids_First_Participant_ID) %>%
+  group_by(tumor_descriptor, experimental_strategy_sum) %>%
+  tally() %>%
+  arrange(desc(n)) %>% 
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
```

Comment:
Getting the following error: 
```
Error in eval(quote(list(...)), env) :
object 'Kids_First_Participant_ID' not found
```

You are losing the info from the above code where you use `summarise`: 
```
assays %>%
  group_by(Kids_First_Participant_ID, match_id) %>%
  summarise(experimental_strategy = str_c(experimental_strategy, collapse = ";"))
```
It is because the `summarise` call creates a new df (https://dplyr.tidyverse.org/reference/summarise.html), so you are only left with two columns: 
```
> names(paired_assays_all)
[1] "experimental_strategy"     "experimental_strategy_sum"
```
This also causes downstream code to break. I suggest you separate out the `summarise` portion (ie notusing the pipe) and add it later if needed.

Reply:
I am creating the main df and then when I create the flextables, I do that separate so I won't miss the information needed later on in the notebook. For the `paired_assays_all`, the df is created in Lines 96-101. Can you please confirm that this is via Docker?